packages/avro-ts/src/convert.ts

```diff
@@ -29,8 +27,30 @@ export const nameParts = (fullName: string): [string] | [string, string] => {
     : [parts[0]];
 };
 
+export const collectRefs = (type: Schema, context: Context): Context => {
+  if (isUnion(type)) {
+    return type.reduce((all, item) => collectRefs(item, all), context);
+  } else if (isArrayType(type)) {
+    return collectRefs(type.items, context);
+  } else if (isMapType(type)) {
+    return collectRefs(type.values, context);
+  } else if (isRecordType(type)) {
+    return type.fields.reduce(
+      (all, item) =>
+        collectRefs(item.type, {
+          ...all,
+          namespace: type.namespace ?? all.namespace,
+          refs: { ...all.refs, [fullName(all, type)]: type },
+        }),
+      context,
+    );
+  } else {
+    return context;
+  }
+};
```

Comment:
I was thinking if there is a nicer way to write this with a switch statement, but I couldn't figure out a way that wouldn't just end up with more code!